web/impact/impact/v1/views/base_list_view.py

```diff
@@ -63,6 +74,41 @@ def _filter_by_date(self, qs, after, before):
 
 def _url(base_url, limit, offset):
     if offset >= 0:
-        return base_url + "?limit={limit}&offset={offset}".format(
-            limit=limit, offset=offset)
+        url = _update_query_param(base_url, "limit", limit)
+        url = _update_query_param(url, "offset", offset)
+        return url
     return None
+
+
+def _base_url(request):
+    absolute_uri = request.build_absolute_uri()
+    absolute_uri = _remove_query_param(absolute_uri, "offset")
+    absolute_uri = _remove_query_param(absolute_uri, "limit")
+    return absolute_uri
+
+
+def _remove_query_param(url, query_param_key):
+    parsed = urlparse(url)
+    qs = parsed.query
+    if not qs:
+        return url
+    parsed_qs = parse_qs(qs, strict_parsing=True)
+    parsed_qs.pop(query_param_key, None)
+    return _build_url_with_query(parsed, parsed_qs)
+
+
+def _update_query_param(url, query_param_key, query_param_value=None):
+    parsed = urlparse(url)
+    qs = parsed.query
+    parsed_qs = parse_qs(qs) if qs else {}
+    parsed_qs.update({query_param_key: query_param_value})
+    return _build_url_with_query(parsed, parsed_qs)
+
+
+def _build_url_with_query(parsed_url, parsed_qs):
+    return urlunparse((parsed_url[0],
```

Comment:
Not a blocker, but I'd rather see these handled using the ParseResult object that actually comes back from urlparse.  It should allow you to just do:
```
parsed_url.query = urlencode(parsed_qs, doseq=True)
urlunparse(parsed_url)
```
rather than using numeric indexes.

Reply:
I tried that and it didn't allow me to do `set_attribute` on it, which was indeed a sad moment.

Reply:
Interesting.  I was almost certain that I got it to work for me in a python shell, but now I can't replicate it.